### PR TITLE
Separate PR and Signed pipelines, migrate to 1ES pipelines templates

### PR DESCRIPTION
**Pull request by vzarytovskii**
We will now have two separate pipeline definitions:

  1. `azure-pipelines.yml` for official signed builds, it will use standard 1ES template, with additional injected checks. This pipeline will build packages, publish them to darc, and trigger VS insertion (only for release branches). It only runs on pushes to branches in the signed pipeline.
  2. `azure-pipelines-PR.yml` for PR builds, this will be running on each PR to main or release branch and run standard set of tests. It runs on both pushes and PRs in public pipeline.

On the higher level nothing should change, other than we'll just have them separate as well as based off of 1ES pipeline templates.

### CODE REVIEW

File: eng/common/SetupNugetSources.ps1 [hunk 1]
```diff
@@ -85,7 +85,7 @@ function AddCredential($creds, $source, $username, $password) {
     $passwordElement.SetAttribute("value", $Password)
 }
 
-function InsertMaestroPrivateFeedCredentials($Sources, $Creds, $Username, $Password) {
+function InsertMaestroPrivateFeedCredentials($Sources, [SecureString] $Creds, $Username, [SecureString] $Password) {
     $maestroPrivateSources = $Sources.SelectNodes("add[contains(@key,'darc-int')]")
 
     Write-Host "Inserting credentials for $($maestroPrivateSources.Count) Maestro's private feeds."
```
Comment (vzarytovskii) [inline eng/common/SetupNugetSources.ps1:91]: These changes are temporary, to unblock us with SDL checking, until Arcade has a fix.

**Comment by vzarytovskii** [comment]
@dotnet/fsharp-team-msft This is ready, the rest of the changes will be small tweaks.

File: eng/common/SetupNugetSources.ps1 [hunk 2]
```diff
@@ -25,7 +25,7 @@
 [CmdletBinding()]
 param (
     [Parameter(Mandatory = $true)][string]$ConfigFile,
-    [Parameter(Mandatory = $true)][string]$Password
+    [Parameter(Mandatory = $true)][SecureString]$Password
```
Comment (vzarytovskii) [inline eng/common/SetupNugetSources.ps1:28]: I will revert this, since Arcade fixed script on their side.

**Review by T-Gro — APPROVED** [review]

File: eng/release/insert-into-vs.yml [hunk 3]
```diff
@@ -69,5 +73,5 @@ stages:
     - task: ms-vseng.MicroBuildShipTasks.55100717-a81d-45ea-a363-b8fe3ec375ad.MicroBuildInsertVsPayload@3
       displayName: 'Insert VS Payload'
       inputs:
-        LinkWorkItemsToPR: false
+        LinkWorkItemsToPR: true
```
Comment (psfinaki) [inline eng/release/insert-into-vs.yml:76]: This is on purpose?

Comment (vzarytovskii) [reply eng/release/insert-into-vs.yml:76]: Yes

File: azure-pipelines.yml [hunk 4]
```diff
@@ -61,833 +52,190 @@ variables:
...
+  repositories:
+  - repository: MicroBuildTemplate
+    type: git
+    name: 1ESPipelineTemplates/MicroBuildTemplate
+    ref: refs/tags/release
+
+extends:
+  template: azure-pipelines/MicroBuild.1ES.Official.yml@MicroBuildTemplate
+  parameters:
+    #featureFlags:
```
Comment (psfinaki) [inline azure-pipelines.yml:70]: Can be removed?

Comment (vzarytovskii) [reply azure-pipelines.yml:70]: No, I'll leave it there to not forget what to uncomment when baselines will be failing.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -25,34 +24,26 @@ trigger:
...
+  # Should **never** be 'main' in dotnet/fsharp/refs/heads/main, since it will start inserting to VS twice.
+  - name: FSharpReleaseBranchName
+    value: 1espm #release/dev17.10
+  # VS Insertion branch name (NOT the same as F# branch)
+  # Should be previous release branch in 'main' and 'main' in release branch
+  # (since for all *new* release branches we insert into VS main and for all *previous* releases we insert into corresponding VS release),
+  # i.e. 'rel/d17.9' in dotnet/fsharp/refs/heads/main and 'main' in F# dotnet/fsharp/refs/heads/release/dev17.10
+  - name: VSInsertionTargetBranchName
+    value: main
   - name: _TeamName
```
Comment (psfinaki) [inline azure-pipelines.yml:39]: This looks duplicated

Comment (vzarytovskii) [reply azure-pipelines.yml:39]: It's not, different purpose

File: azure-pipelines.yml [hunk 6]
```diff
@@ -61,833 +52,190 @@ variables:
...
+              value: Real
+            steps:
+            - checkout: self
+              clean: true
+            - template: /eng/restore-internal-tools.yml
+            - script: eng\CIBuild.cmd
+                      -configuration $(_BuildConfig)
+                      -prepareMachine
+                      -testAllButIntegration
+                      -officialSkipTests $(SkipTests)
```
Comment (psfinaki) [inline azure-pipelines.yml:136]: SkipTests doesn't seem to be initialized anywhere, probably can be removed.

Comment (vzarytovskii) [reply azure-pipelines.yml:136]: It can't it's passed in CI

**Review by psfinaki — APPROVED** [review]

**Review by KevinRansom — APPROVED** [review]

**Comment by vzarytovskii** [comment]
This one succeeds full cycle - PR builds, signed builds, insertion into SDK and into VS. 
I am going to merge it, since we're closing to deadline, and all other fixes (if any) will be addressed separately if different PRs.

**Pull request merged**